torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -905,7 +905,8 @@ std::string toString(SegmentedFusion* segmented_fusion) {
 std::unique_ptr<Fusion> SegmentedFusion::makeFusion(SegmentedGroup* sg) {
   std::unique_ptr<Fusion> fusion_segment = std::make_unique<Fusion>();
 
-  auto complete_to_segment_map = Fusion::copy(&fusion_, fusion_segment.get());
+  auto complete_to_segment_map =
+      Fusion::copy(completeFusion(), fusion_segment.get());
```

Comment:
Is it time to get some expr sorting in this printing? If I recall correctly the output exprs are unsorted. You could do something similar in keeping ordering by knowing when we merge two groups together to understand producer-consumer relationship. Though I'm uncertain how that may work when doing horizontal fusions in combination with vertical.

Reply:
Sure. I will added expr sorting in the printing in this PR.

Reply:
Added sorting in the prints. Now the order groups are printed and exprs within each group are topologically sorted. The edges are sorted based on the topological order of starting point.